data_structures/binary_tree/binary_tree_traversals.py

```diff
@@ -0,0 +1,191 @@
+class Node:
+    """
+    A Node has data variable and pointers
+    to its left and right nodes.
+    """
+
+    def __init__(self, data):
+        self.left = None
+        self.right = None
+        self.data = data
+
+
+def make_tree() -> Node:
+    root = Node(1)
+    root.left = Node(2)
+    root.right = Node(3)
+    root.left.left = Node(4)
+    root.left.right = Node(5)
+    return root
+
+
+def preorder(root):
+    """
+    PreOrder traversal: visit root node
+    then its left subtree followed
+    by right subtree.
+
+    >>> preorder(make_tree())
+    1 2 4 5 3

```

Comment:
```suggestion
    1 2 4 5 3 
```
That extra space at the end is needed.  #3306 will fix the pre-commit failure.  Sorry this is so hard but this PR is really helping us to debug our testing